streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
Actually I think the previous commit is correct. Why do you want to reverse the before and after semantics?

More specifically, when we call:

```
stream1.join(stream2, JoinWindows.of(0).before(100))
```

And according to our java doc string: 

```
Specifies that records of the same key are joinable if their timestamps are within
the join window interval, and if the timestamp of a record from the secondary stream is
earlier than or equal to the timestamp of a record from the first stream.
```

It means that a record from `stream1` with time `1000` is joinable with `stream2`'s time range `900 - 1000`; and a record from `stream2` with time `1000` is joinable with `stream1`'s time range `1000 - 1100`. Right?